content/docs/migration_1.0/Concepts.md

```diff
@@ -0,0 +1,159 @@
+
+<!-- [key expressions](#key-expression) -->
+Each Language bindings will have their own specifics of Serializing and Deserializing, but for the most part it will involve implementing a serialize / deserialize function for your datatype or make use of auto-generated conversions for composite types.
+
+## Encoding
+
+`Encoding` has been reworked, moving away from enumerables to now accepting strings.
+While Zenoh does not impose any `Encoding` requirement on the user, providing an `Encoding` can offer automatic wire-level optimization for known types.  
+For the user defined `Encoding`, it can be thought of as optional metadata, carried over by Zenoh in such a way that the end user’s application may perform different operations based on `Encoding`.  
+We have expanded our list of predefined encoding types from Zenoh 0.11.0 to include variants for numerous IANA standard encodings, including but not limited to  `video/x` , `application/x`, `text/x`, `image/x` and `audio/x` encoding families, as well as an encoding family specific to Zenoh defined by the prefix `zenoh/x` .   
+Users can also define their own encoding scheme that does not need to be based on the predefined IANA variants. Upon declaring a specific encoding, the users encoding scheme will be prefixed with `zenoh/bytes` for distinction.
+
+
+## Attachment
+
+We have made attachment more flexible across API’s, essentially accepting anything that can be converted to a `ZBytes` as an optional extra to `put` , `delete` , on `Query`'s and Query `reply`'s.  
+We also allow for composite types to be converted into `ZBytes`, meaning that using the Attachment API as a metadata transport is easier than ever.
+
+## Query & Queryable
+
+The `reply` method of a `Queryable` has gained two variants: `reply_del` and `reply_err` to respectively indicate that a deletion should be performed and that an error occurred.   
+Additionally, the 3 variants behave similarly to `put` and `del`, hence providing improved ergonomics.
+
+We have added the ability to get the underlying `Handler` of a Queryable as well.
+
+## Use accessors to get private members
+Across language bindings we encapsulate members of structs, and they can’t be accessed directly now.  
+The only way to access struct values is to use the getter function associated with them.
+
+
+## Pull Subscribers have been removed
+
+The concept of a pull subscriber no longer exists in Zenoh.
+However, when creating a `Subscriber`, it may be the case that developers only care about the latest data and want to discard the oldest data. 
+The `RingChannel` can be used to get a similar behaviour.
+This contrasts with the `FIFOChannel`, the default channel type used internally in Subscribers, which drops new message once its buffer is full.
+You can take a look at examples of usage in any language’s examples/z_pull.x
```

Comment:
Could we add a link to the various examples?

Reply:
Added example for `RingChannel`
There is only an example for `RingChannel` 
The user will be able to figure out to put `FifoChannel` instead of `RingChannel` in when declaring a subscriber.